Objects/exceptions.c

```diff
@@ -2862,6 +2862,51 @@ _PyErr_TrySetFromCause(const char *format, ...)
  * or minus, using the stream redirection syntax).
  */
 
+
+// Static helper for setting legacy print error message
+static int
+_set_legacy_print_statement_msg(PySyntaxErrorObject *self, Py_ssize_t start)
+{
+
+    PyObject *strip_sep_obj = PyUnicode_FromString("\r\n");
+    Py_UCS4 soft_space_check = PyUnicode_READ_CHAR(PyUnicode_FromString(","), 0);
+
+    if (strip_sep_obj == NULL)
+        return -1;
+
+    // PRINT_OFFSET is to remove `print ` word from the data.
+    const int PRINT_OFFSET = 6;
+    Py_ssize_t text_len = PyUnicode_GET_LENGTH(self->text);
+    PyObject *data = PyUnicode_Substring(self->text, PRINT_OFFSET, text_len);
+
+    if (data == NULL)
+        return -1;
+
+    data = _PyUnicode_XStrip(data, 1, strip_sep_obj);
```

Comment:
The result of `_PyUnicode_XStrip()` is not checked for error. Old value of `data` is leaked.